adapters/repos/db/vector/hnsw/commit_logger.go

```diff
@@ -441,48 +457,58 @@ func (l *hnswCommitLogger) RootPath() string {
 	return l.rootPath
 }
 
-func (l *hnswCommitLogger) startSwitchLogs(stopFunc cyclemanager.StopFunc) {
-	if err := l.SwitchCommitLogs(false); err != nil {
+func (l *hnswCommitLogger) startSwitchLogs(shouldBreak cyclemanager.ShouldBreakFunc) bool {
+	err, executed := l.switchCommitLogs(false)
+	if err != nil {
 		l.logger.WithError(err).
 			WithField("action", "hnsw_commit_log_maintenance").
 			Error("hnsw commit log maintenance failed")
 	}
+	return executed
 }
 
-func (l *hnswCommitLogger) startCombineAndCondenseLogs(stopFunc cyclemanager.StopFunc) {
-	if err := l.combineLogs(); err != nil {
+func (l *hnswCommitLogger) startCombineAndCondenseLogs(shouldBreak cyclemanager.ShouldBreakFunc) bool {
+	err, executed1 := l.combineLogs()
+	if err != nil {
 		l.logger.WithError(err).
 			WithField("action", "hnsw_commit_log_combining").
 			Error("hnsw commit log maintenance (combining) failed")
 	}
 
-	if err := l.condenseOldLogs(); err != nil {
+	err, executed2 := l.condenseOldLogs()
+	if err != nil {
 		l.logger.WithError(err).
 			WithField("action", "hnsw_commit_log_condensing").
 			Error("hnsw commit log maintenance (condensing) failed")
 	}
+	return executed1 || executed2
 }
 
 func (l *hnswCommitLogger) SwitchCommitLogs(force bool) error {
+	err, _ := l.switchCommitLogs(force)
+	return err
+}
+
+func (l *hnswCommitLogger) switchCommitLogs(force bool) (error, bool) {
```

Comment:
shouldn't it be in the return statement (bool, error)? instead of (error, bool)?